needle-zendesk-connector/src/server/backend/connectors-backend.ts

```diff
@@ -0,0 +1,257 @@
+
+  // Process tickets
+  for (const ticket of params.selectedTickets) {
+    const ticketFileId = createNeedleFileId();
+    const commentsFileId = createNeedleFileId();
+
+    fileIdMap.set(`ticket-${ticket.id}`, ticketFileId);
+    fileIdMap.set(`comments-${ticket.id}`, commentsFileId);
+
+    descriptor.create.push(
+      {
+        id: ticketFileId,
+        url: ticket.url,
+        type: "text/plain",
+      },
+      {
+        id: commentsFileId,
+        url: ticket.url.replace(".json", "/comments.json"),
+        type: "text/plain",
+      },
+    );
+  }
+
+  // Process articles
+  for (const article of params.selectedArticles) {
+    const articleFileId = createNeedleFileId();
+    fileIdMap.set(`article-${article.id}`, articleFileId);
+
+    descriptor.create.push({
+      id: articleFileId,
+      url: article.html_url,
+      type: "text/html",
+    });
+  }
+
+  await db.transaction(async (tx) => {
+    await tx.insert(zendeskConnectorsTable).values({
+      connectorId: connector.id,
+      subdomain: params.subdomain,
+      orgId: params.organizationId,
+      includeTickets: params.selectedTickets.length > 0,
+      includeArticles: params.selectedArticles.length > 0,
+    });
+
+    await handleDatabaseUpdates(
+      tx,
+      connector.id,
+      { create: params.selectedTickets, update: [], delete: [] },
+      { create: params.selectedArticles, update: [], delete: [] },
+      fileIdMap,
+    );
+  });
+
+  await publishConnectorRun(connector.id, descriptor);
+  return connector;
+}
+
+export async function runZendeskConnector(
+  { connectorId }: { connectorId: string },
+  session?: Session,
+) {
+  let accessToken = "";
+  if (session) {
+    const { credentials } = await getConnector(connectorId, session.id);
+    accessToken = credentials ?? "";
+  }
+
+  const connectorDetails = await getConnectorDetails(connectorId);
+  if (!connectorDetails) {
+    throw new Error(`No Zendesk connector found for ID: ${connectorId}`);
+  }
+
+  const currentFiles = await getCurrentFiles(connectorId);
+  const currentTickets = currentFiles.filter(
+    (f) => f.type === "ticket" || f.type === "comments",
+  ) as DbFile[];
+  const currentArticles = currentFiles.filter(
+    (f) => f.type === "article",
+  ) as DbFile[];
+
+  const zendeskService = createZendeskService(
+    accessToken,
+    connectorDetails.subdomain,
+  );
+  const { tickets: liveTickets, articles: liveArticles } =
+    await zendeskService.searchByOrganization(
+      connectorDetails.orgId.toString(),
+      {
+        fetchTickets: connectorDetails.includeTickets,
+        fetchArticles: connectorDetails.includeArticles,
+      },
+    );
+
+  const ticketsDiff = liveTickets
+    ? computeDiff(currentTickets, liveTickets.items)
+    : { create: [], update: [], delete: currentTickets };
+  const articlesDiff = liveArticles
+    ? computeDiff(currentArticles, liveArticles.items)
+    : { create: [], update: [], delete: currentArticles };
```

Comment:
nit: i find it cleaner to organize the code in such a way:

```js
const liveTicketItems = liveTickets?.items ?? []
const diffActions1 = computeDiff(currentTickets, liveTicketItems)

const liveArticlesItems = liveArticles?.items ?? []
const diffActions2 = computeDiff(currentArticles, liveArticleItems)
```

Reply:
Thanks for the suggestion, but I think we can't simplify it that way. The current conditional logic is intentional:
When `liveTickets` is null, we want to delete ALL current tickets (not just compute a diff against an empty array). The same applies for articles.
The suggestion would change the behavior to always compute a diff (even against an empty array), which would miss our "delete everything" fallback case.

Reply:
ah i see, you want to skip calling `computeDiff` if we know we'll delete all... as a compute optimization 👍🏻